src/ui/components/Library/Team/View/NewTestRuns/TestRunResultList.tsx

```diff
@@ -0,0 +1,50 @@
+import { useContext } from "react";
+
+import { TestRunTestWithRecordings } from "shared/test-suites/TestRun";
+
+import { useTestRunDetailsSuspends } from "../TestRuns/hooks/useTestRunDetailsSuspends";
+import { Alert } from "./Alert";
+import { TestResultListItem } from "./Overview/TestResultListItem";
+import { TestRunsContext } from "./TestRunsContextRoot";
+
+export function TestRunResultList({
+  selectedSpecTests,
+}: {
+  selectedSpecTests: TestRunTestWithRecordings[];
+}) {
+  const { testRunId } = useContext(TestRunsContext);
+  const { testRun } = useTestRunDetailsSuspends(testRunId);
+  const hasRecordings = selectedSpecTests.some(test =>
```

Comment:
This is probably the code I wrote but the logic is a little convoluted to have two `some`'s and then `!hasRecordings`.

Instead, maybe:
```
const noRecordings = selectedSpecTests.all(test => test.executions.all(e => e.recordings.length === 0))
if (noRecordings) {
  return ...
}
```

Reply:
`.some` and `.map` are both iterating over the entire collection. We only need one `.forEach` and then it can conditionally push to a derived array.

Reply:
I think that's a bit more readable — went ahead and applied it.